drivers:ethernet: Add promiscuous mode to the slip interface

mrjimenez opened the pull request:
Slip is naturally promiscuous, so this patch does nothing but acknowledge that. Promiscuous mode in slip is important to allow the interface to be added to a bridge.

jukkar reviewed (commented):
You have mixed changes in the commit. The indentation/style things should be separated into a different commit than the promisc mode changes.

mrjimenez committed:
drivers: ethernet: Run clang-format before applying a patch

This patch just formats the file before the real patch so that
whitespace changes do not mix with the real code changes.

Signed-off-by: Marcelo Roberto Jimenez <marcelo.jimenez@gmail.com>

mrjimenez committed:
drivers: ethernet: Add promiscuous mode to slip interface

Slip is naturally promiscuous, so this patch does nothing but
acknowledge that. Promiscuous mode in slip is important to allow
the interface to be added to a bridge.

Signed-off-by: Marcelo Roberto Jimenez <marcelo.jimenez@gmail.com>
